packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -217,124 +179,77 @@ const DataGrid = injectIntl((props) => {
-  }, [hasSelectableRows]);
-
-  // useEffect for row displayed columns
-  useEffect(() => {
-    setDataGridColumns(displayedColumns.map((column) => initializeColumn(column)));
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [pinnedColumns, overflowColumns]);
-
-  // useEffect to calculate pinned column offsets
-  useEffect(() => {
-    const offsetArray = [];
-    let cumulativeOffset = 0;
-    let lastPinnedColumnIndex;
-
-    // if grid has selectable rows but no pinned columns, then set the offset of the first column to 0
-    if (hasSelectableRows && pinnedColumns.length === 0) {
-      lastPinnedColumnIndex = 0;
-      offsetArray.push(cumulativeOffset);
-      setPinnedColumnOffsets(offsetArray);
-      return;
+    if (gridContainerRef) {
+      const table = gridContainerRef.current.getElementsByTagName('table')[0];
+      tableRef.current = table;
     }
-
-    if (pinnedColumns.length > 0) {
-      offsetArray.push(cumulativeOffset);
-
-      lastPinnedColumnIndex = hasSelectableRows ? pinnedColumns.length : pinnedColumns.length - 1;
-
-      // eslint-disable-next-line array-callback-return
-      dataGridColumns.slice(0, lastPinnedColumnIndex).map((pinnedColumn) => {
-        cumulativeOffset += pinnedColumn.width;
-        offsetArray.push(cumulativeOffset);
-      });
-    }
-    setPinnedColumnOffsets(offsetArray);
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [dataGridColumns]);
+  }, [gridContainerRef]);
 
   // -------------------------------------
 
   const handleMoveCellFocus = (fromCell, toCell) => {
-    // Obtain coordinate rectangles for grid container, column header, and new cell selection
-    const gridContainerRect = gridContainerRef.current.getBoundingClientRect();
-    const columnHeaderRect = grid.current.rows[0].cells[toCell.col].getBoundingClientRect();
-    const nextCellRect = grid.current.rows[toCell.row].cells[toCell.col].getBoundingClientRect();
-
-    // Calculate horizontal scroll offset for right boundary
-    if (nextCellRect.right > gridContainerRect.right) {
-      gridContainerRef.current.scrollBy(nextCellRect.right - gridContainerRect.right, 0);
-    } else {
-      // Calculate horizontal scroll offset for left boundary
-      let scrollOffsetX = 0;
-      if (pinnedColumnOffsets.length > 0) {
-        if (toCell.col > pinnedColumnOffsets.length - 1) {
-          const lastPinnedColumnRect = grid.current.rows[toCell.row].cells[pinnedColumnOffsets.length - 1].getBoundingClientRect();
-          scrollOffsetX = nextCellRect.left - lastPinnedColumnRect.right;
-        }
-      } else {
-        scrollOffsetX = nextCellRect.left - gridContainerRect.left;
-      }
-
-      if (scrollOffsetX < 0) {
-        gridContainerRef.current.scrollBy(scrollOffsetX, 0);
+    // Obtain coordinate rectangles for scrollable container, column header, and new cell selection
+    // TODO: Implementation is not easy to understand, get suggestions on how to clean this logic up.
```

Comment:
We should not be leaving a TODO in our code.

Reply:
Took this out, this was doing too much.